### PR TITLE
Move r_print.h into r_util/r_print.h ##refactor

### LINKED ISSUE #11918: Move r_print.h into r_util/r_print.h

### PR DESCRIPTION
**Pull request by ar7z1**
In this PR I've fixed #11918: moved r_print.h into r_util/r_print.h.

### CODE REVIEW

File: binr/rafind2/rafind2.c [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #include <getopt.c>
 
 #include <r_types.h>
-#include <r_print.h>
+#include <r_util/r_print.h>
```
Comment (radare) [inline binr/rafind2/rafind2.c]: i think that this include should be inside r_util.h, so we can get rid of all those lines

Comment (ar7z1) [reply binr/rafind2/rafind2.c]: @radare Ok, I'll fix it. :-)

**Comment by radare** [comment]
eta wen

File: libr/include/r_cons.h [hunk 2]
```diff
@@ -6,7 +6,17 @@ extern "C" {
 #endif
 
 #include <r_types.h>
-#include <r_util.h>
```
Comment (ar7z1) [inline libr/include/r_cons.h]: `r_print.h` uses `r_cons.h`

File: libr/include/r_io.h [hunk 3]
```diff
@@ -4,11 +4,15 @@
...
+#include "r_util/r_assert.h"
+#include "r_util/r_idpool.h"
+#include "r_util/r_cache.h"
+#include "r_util/r_buf.h"
+#include "r_util/r_itv.h"
+#include "r_util/r_rbtree.h"
+#include "r_util/r_num.h"
+#include "r_util/r_file.h"
 #include "r_socket.h"
-#include "r_util.h"
```
Comment (ar7z1) [inline libr/include/r_io.h]: `r_print.h` uses `r_io.h`

File: libr/include/r_reg.h [hunk 4]
```diff
@@ -2,7 +2,8 @@
 #define R2_REG_H
 
 #include <r_types.h>
-#include <r_util.h>
```
Comment (ar7z1) [inline libr/include/r_reg.h]: `r_print.h` uses `r_reg.h`

File: libr/include/r_util/r_idpool.h [hunk 5]
```diff
@@ -3,8 +3,6 @@
 #ifndef R_ID_STORAGE_H
 #define R_ID_STORAGE_H
 
-#include <r_util/r_pool.h>
```
Comment (ar7z1) [inline libr/include/r_util/r_idpool.h]: Already exists in `r_util.h`

File: binr/ragg2/ragg2.c [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 #include <r_egg.h>
 #include <r_bin.h>
-#include <r_print.h>
+#include <r_util.h>
```
Comment (radare) [inline binr/ragg2/ragg2.c]: this is already included in r_bin.h

**Comment by radare** [comment]
please rebase again

**Comment by ar7z1** [comment]
@radare Done! :-)

**Comment by radare** [comment]
;-)

**Comment by ret2libc** [comment]
Mmm is this red just because of rebase?

**Comment by ar7z1** [comment]
@ret2libc I don't think so. I had a long flight, now I'm at home and hope to debug and fix it. :-)

**Comment by radare** [comment]
5min build times seems like a compilation problem. but yeah, rebasing and squashing will make changes cleaner and probably solve a test

**Comment by radare** [comment]
Please rebase

**Comment by radare** [comment]
Ping

**Comment by radare** [comment]
please resolve the conflict and rebase

**Comment by radare** [comment]
the XX in travis makes no sense to me wtf

File: libr/include/r_io.h [hunk 7]
```diff
@@ -4,11 +4,16 @@
 #define R2_IO_H
 
 #include "r_list.h"
-#include <r_util/r_idpool.h>
-#include <r_util/r_cache.h>
-#include <r_util/r_buf.h>
+#include "r_util/r_assert.h"
```
Comment (radare) [inline libr/include/r_io.h]: why not just include r_util.h ? instead of being that picky?

File: libr/include/r_io.h (hunk 3 above)
Comment (radare) [reply libr/include/r_io.h]: ok nvm :D

**Comment by radare** [comment]
i'll merge and solve the issues in master if any, the pr lgtm

**Comment by radare** [comment]
Dude.... i didnt checked out the warnings... i cant understand how you can fix something without seeing there are pages and pages of warnings introduced by your changes. Next time check them out. they are there for a reason. im fixing the issues in  master now

**Comment by ar7z1** [comment]
@radare Sorry, I worked on it in my fork. I didn't know that you merge this PR with failed build. :-(
This PR needs a lot of fixes (huge number of warnings - one of them).
I think the best way is to completely revert the merge and not to fix master in a hurry. And then I finish, I'll recreate PR.

**Pull request merged**